More reactive SQL client connection options

tsegismont opened the pull request:
Added reconnectAttempts, reconnectInterval and idleTimeout props to reactive datasource config	

Resolves #11149

tsegismont committed:
Added reconnectAttempts, reconnectInterval and idleTimeout props to reactive datasource config

gsmet commented on extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:
```diff
@@ -94,4 +95,22 @@
+    /**
+     * The maximum time without data written to or read from a connection before it is removed from the pool.
+     */
+    @ConfigItem(defaultValue = "PT0S")
```
Hmmm, it looks like this one should more be `Optional`?

tsegismont replied:
Done

gsmet reviewed (changes requested):
Added a suggestion. Happy to discuss it as I'm not really sure.

gsmet commented:
@machi1990 we have a doc generation issue with the duration note. We include it twice in the datasource.doc now that we have it included here too. I don't exactly know how we could fix that as these things are generated totally independently.

gsmet commented:
I will try to figure out how to fix the doc issue.

tsegismont commented:
Thanks

Le jeu. 20 août 2020 à 16:07, Guillaume Smet <notifications@github.com> a
écrit :

> I will try to figure out how to fix the doc issue.
>

gsmet committed:
Allow to disable the addition of the duration format note

This is useful when you have several config references included in the
same document and you only want one note.

We can probably do better but I prefer to have a quick fix to unblock
this PR.

gsmet commented:
I added a quick (and dirty) fix for the doc issue. Let's see what CI says.

machi1990 commented:
> @machi1990 we have a doc generation issue with the duration note. We include it twice in the datasource.doc now that we have it included here too. I don't exactly know how we could fix that as these things are generated totally independently.

@gsmet @tsegismont sorry only seeing this comment now. Good thing we had a fix in place :-)
